### DISCUSSION TITLE
additional menu like the m5 dial example

### DISCUSSION THREAD

**Discussion by danielscheidler** [SmartHome-yourself/m5-dial-for-esphome#5] [Ideas]
Perhaps it would be nice to add an additional menü like the dial example to switch between the devices. 
Especially when many devices have been added, wiping quickly becomes annoying otherwise.

Perhaps you have further ideas for functions in such a menu?

**Comment by ArduFox** [comment]
settings would be nice, like switching clicking sounds on / off or dimming the display.

**Comment by ArduFox** [comment]
Ich habe einen ersten Prototyp eines runden Menüs im Layout einer Uhr, das noch keine weitere Funktionalität hat. Das Menü habe ich als neues Device inkl. mode aufgebaut. 
Aktuell arbeite ich an Details und an der Möglichkeit auf das array der devices zuzugreifen bzw. ShysM5Dial um Methoden für einen sicheren Zugriff darauf zu erweitern. Dann kann ich einen direkten Wechsel zu einem ausgewählten device realisieren.
Vermutlich werde ich noch an Ostern oder am Wochenende danach einen push in den dev-code machen

**Comment by danielscheidler** [comment]
Klingt an sich super. Ich weiß nur nicht ob ich das Menü als device
aufbauen würde. Die ha Anbindung etc wird ja eh nicht benötigt. Glaube das
würde ich eher als eigenständige Klasse erstellen und im Main
initialisieren. Dann kann man es über callbacks auch direkt an die
setCurrentDevice Methode oä anbinden


ArduFox ***@***.***> schrieb am So., 31. März 2024, 13:50:

**Reply by ArduFox** [reply]
Ja das hatte ich auch überlegt aber für mich war es im ersten Schritt einfacher die Device Struktur zu nutzen und mit Swipe ins Menü zu wechseln. Außerdem kann dann im Konfigurations.yaml selbst eingestellt werden, ob es ein Menü geben soll und welche modes dafür genutzt werden.
Allerdings hatte ich einige Mühe mit der Reihenfolge der includes und der Klassen-Deklarationen und Implementierungen. Läuft jetzt aber gefällt mir noch nicht und Dir wahrscheinlich noch weniger.
